web3/_utils/events.py

```diff
@@ -193,7 +196,7 @@ def get_event_abi_types_for_decoding(event_inputs: Sequence[ABIEventParams]) ->
         if input_abi['indexed'] and is_dynamic_sized_type(input_abi['type']):
             yield 'bytes32'
         else:
-            yield input_abi['type']
+            yield collapse_if_tuple(dict(input_abi))
```

Comment:
This is sort of nitpicky, but do you think it's worth making this `collapse_if_tuple(dict(...))` a function of it's own?

Reply:
Yeah... this might make the most sense. Something that describes the type-aware nature of the parsing without being specific to tuple. Not sure what to call this method yet but I do think it could be abstracted at least within the web3.py utils.

Any thoughts on a name for the method?

Reply:
def get_normalized_abi_arg_type(abi_arg: ABIEventParams) -> str:
    """
    Return the normalized type for the abi argument provided. In order to account for tuple argument
    types, this abstraction makes use of `collapse_if_tuple()` to collapse the appropriate component
    types within the tuple type.
    """
    return collapse_if_tuple(dict(abi_arg))